File: generate.js
const fs = require('fs');
const path = require('path');

const config = JSON.parse(fs.readFileSync('./redirects.json', 'utf8'));
const outDir = './dist';

if (fs.existsSync(outDir)) {
  fs.rmSync(outDir, { recursive: true, force: true });
}
fs.mkdirSync(outDir, { recursive: true });

for (const [route, target] of Object.entries(config)) {
  let cleanRoute = route.replace(/^\/+/, '').replace(/\/+$/, '');
  const targetDir = path.join(outDir, cleanRoute);
  const targetFile = path.join(targetDir, 'index.html');

  const html = `<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <meta http-equiv="refresh" content="0; url=${target}">
  <title>Redirecting...</title>
</head>
<body>
  <p>Redirecting to <a href="${target}">${target}</a></p>
</body>
</html>`;

  fs.mkdirSync(targetDir, { recursive: true });
  fs.writeFileSync(targetFile, html, 'utf8');
  console.log('Generated ' + targetFile + ' -> ' + target);
}

console.log('All redirects generated successfully.');

File: generate.py
import json, os, shutil

config = json.load(open('./redirects.json', encoding='utf-8'))
out_dir = './dist'

if os.path.exists(out_dir):
    shutil.rmtree(out_dir)
os.makedirs(out_dir, exist_ok=True)

html_template = '''<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <meta http-equiv="refresh" content="0; url={target}">
  <title>Redirecting...</title>
</head>
<body>
  <p>Redirecting to <a href="{target}">{target}</a></p>
</body>
</html>'''

for route, target in config.items():
    clean = route.strip('/')
    target_dir = os.path.join(out_dir, clean)
    os.makedirs(target_dir, exist_ok=True)
    with open(os.path.join(target_dir, 'index.html'), 'w', encoding='utf-8') as f:
        f.write(html_template.format(target=target))
    print(f'Generated {target_dir}/index.html -> {target}')

print('All redirects generated successfully.')